adapter/internal/messaging/azure_listener.go

```diff
@@ -19,32 +19,42 @@
 package messaging
 
 import (
+	"time"
 	"github.com/wso2/product-microgateway/adapter/config"
 	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
-	msg "github.com/wso2/product-microgateway/adapter/pkg/messaging"
-	servicebus "github.com/Azure/azure-service-bus-go"
 	"github.com/wso2/product-microgateway/adapter/pkg/health"
-	"time"
+	msg "github.com/wso2/product-microgateway/adapter/pkg/messaging"
 )
 
 const (
-	componentName string = "adapter"
-	subscriptionIdleTimeDuration = time.Duration(72 * time.Hour)
+	componentName                              = "adapter"
+	subscriptionIdleTimeDuration               = time.Duration(72 * time.Hour)
+	defaultReconnectRetryCount                 = 60
+	defaultReconnectInterval     time.Duration = 5000
 )
 
 // InitiateAndProcessEvents to pass event consumption
 func InitiateAndProcessEvents(config *config.Config) {
 	var err error
-	var namespace *servicebus.Namespace
+	var reconnectRetryCount = config.ControlPlane.ASBConnectionParameters.ReconnectRetryCount
+	var reconnectInterval = config.ControlPlane.ASBConnectionParameters.ReconnectInterval
 	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Starting InitiateAndProcessEvents method")
 	logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] EventListeningEndpoint is ",
 		config.ControlPlane.ASBConnectionParameters.EventListeningEndpoint)
-	namespace, availableTopicList, err := msg.InitiateBrokerConnectionAndGetAvailableTopics(
-		config.ControlPlane.ASBConnectionParameters.EventListeningEndpoint)
+	if reconnectRetryCount == 0 {
+		reconnectRetryCount = defaultReconnectRetryCount
+	}
+	if reconnectInterval == 0 {
+		reconnectInterval = defaultReconnectInterval
```

Comment:
In other places this not how we set default values. The struct is set the default values first and then we override from the config. So you don't have to check for if it is equal to zero

Reply:
ok this was done in earlier broker impl. Will remove it

Reply:
fixed in https://github.com/wso2/product-microgateway/pull/2196/commits/f6486f812ca86d1d6756e920332a21758e1e99c6